build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -124,20 +125,29 @@ private static boolean doesNotHaveCatOperation(ObjectNode doNodeValue) {
         return true;
     }
 
-    public void skipTest(String testName, String reason) {
+    public void skipTest(String fullTestName, String reason) {
         //The tests are defined by 3 parts a/b/c where
         // a = the folder name
         // b = the file name without the .yml extension
-        // c = the root name inside the .yml
+        // c = the test name inside the .yml
         // For example: indices.get_mapping/20_missing_type/Non-existent type returns 404
-        String[] testParts = testName.split("/");
+        // However, the folder can be arbitrarily nest so, a == a1/a2/a3, and the test name can include forward slashes, so c == c1/c2/c3
+        // So we also need to support a1/a2/a3/b/c1/c2/c3
+
+        String[] testParts = fullTestName.split("/");
         if(testParts.length < 3 ){
-            throw new IllegalArgumentException("To skip tests, all 3 parts [folder/file/test name] must be defined. found [" + testName + "]");
+            throw new IllegalArgumentException("To skip tests, all 3 parts [folder/file/test name] must be defined. found [" + fullTestName + "]");
         }
 
-        //some tests have a "/" in the name, so allow for 1 forward slash
-        skippedTestByTestNameTransformations.computeIfAbsent(Pair.of(testParts[0], testParts[1] + ".yml"),
-            k -> new HashMap<>()).put(testParts[2] + (testParts.length == 4 ? "/" + testParts[3] : ""), reason);
+        PatternSet skippedPatternSet = patternSetFactory.create();
+        //create file patterns for all a1/a2/a3/b.yml possibilities.
+        for(int i = testParts.length - 1; i > 1; i-- ){
```

Comment:
Does this have any noticeable impact on performance? We are adding a _ton_ of patterns to the copyspec here.

Reply:
I didn't notice any anidotely, and via the [build scan for this PR](https://gradle-enterprise.elastic.co/s/4obnezggs2yb2/timeline?details=5hulgjlqioq76&hide-timeline) all of the transformations took 1.119 seconds vs. a [build scan for a random PR](https://gradle-enterprise.elastic.co/s/nh6a6j6tbxy6o/timeline?details=5hulgjlqioq76&hide-timeline) without this change took 1.439s.  I assume this work of pattern matching would be the expensive part that should be included in the task timing. 

> We are adding a ton of patterns to the copyspec here.

It shouldn't be a ton. It will generally be 1 include pattern per skipTest. 2 if there there 4 parts, 3 if there are 5 parts, and so on. That 1 (maybe 2 or 3) include patterns will then match against the FileTree of the tests N times, where N = number of defined skipTest's per build.gradle.

For :rest-api-spec it will match the tests FileTree against 1 include pattern ~30 times, and then match 2 include patterns 1 time (for the special case where we actually 4 parts).

Reply:
Sounds good. This is within the margin of error I think.